build.gradle

```diff
@@ -40,28 +40,23 @@ repositories {
 }
 
 dependencies {
-    // Common utils
-    implementation group: 'org.apache.commons', name: 'commons-dbcp2', version: '2.11.0'
-    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.14.0'
-    implementation group: 'org.apache.commons', name: 'commons-pool2', version: '2.12.0'
+    // Common utils (unversioned are versioned by Spring dependency manager)
```

Comment:
It looks like there are now many cases where versions are provided by spring. Do you think we need a comment on each case? If a new dependency is added without a comment, do you think it will be confusing for future readers?

I was thinking that, in general, we should have good tools to help us determine where a version is coming from, so if the need arises we can figure it out, rather than trying to comment each instance individually. But maybe you find this approach better?

Reply:
Good question.  The comments throughout felt like overkill as I was writing them and I agree this could get out of date.  But for this repo, this was a pretty big shift in approach, and I have found that analyzing dependencies is not yet a commonly-held competency (I have intentionally built this up for myself over the past few months).

I'll replace the comments throughout with a top-level comment though.